registry/remote/repository.go

```diff
@@ -361,38 +436,86 @@ func (r *Repository) referrers(ctx context.Context, artifactType string, fn func
 	if resp.StatusCode != http.StatusOK {
 		return "", errutil.ParseErrorResponse(resp)
 	}
-	if !legacyAPI {
-		if err := verifyOrasApiVersion(resp); err != nil {
-			return "", err
-		}
-	}
 
-	var page struct {
-		References []ocispec.Descriptor `json:"references"`
-		Referrers  []ocispec.Descriptor `json:"referrers"`
-	}
+	var index ocispec.Index
 	lr := limitReader(resp.Body, r.MaxMetadataBytes)
-	if err := json.NewDecoder(lr).Decode(&page); err != nil {
+	if err := json.NewDecoder(lr).Decode(&index); err != nil {
 		return "", fmt.Errorf("%s %q: failed to decode response: %w", resp.Request.Method, resp.Request.URL, err)
 	}
-	var refs []ocispec.Descriptor
-	if legacyAPI {
-		refs = page.References
-	} else {
-		refs = page.Referrers
+	referrers := index.Manifests
+	if artifactType != "" &&
+		!isReferrersFilterApplied(index.Annotations, "artifactType") {
```

Comment:
nit: it is actually more readable if it is one line. It won't be longer than line 443.
```suggestion
	if artifactType != "" && !isReferrersFilterApplied(index.Annotations, "artifactType") {
```